code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,265 @@
+	. = ..()
+	if (!core_installed)
+		. += span_warning("It requires a hallucination anomaly core in order to function.")
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_HEAD
+
+/obj/item/clothing/head/helmet/perceptomatrix/update_icon_state()
+	icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	worn_icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_interaction(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/hallucination))
+		return NONE
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return ITEM_INTERACT_BLOCKING
+	qdel(weapon)
+	core_installed = TRUE
+	update_anomaly_state()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/crate/crate_open.ogg', 50, FALSE)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/clothing/head/helmet/perceptomatrix/functioning
+	core_installed = TRUE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination
+	name = "Hallucinate"
+	desc = "Redirect perceptual energies towards a target, staggering them."
+	button_icon_state = "blind"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/blind_target.dmi'
+
+	sound = 'sound/items/weapons/emitter2.ogg'
+	school = SCHOOL_PSYCHIC
+	cooldown_time = 15 SECONDS
+
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+
+	active_msg = "You prepare to zap a target with hallucinations..."
+
+	/// The amount of blurriness to apply
+	var/eye_blur_duration = 7 SECONDS
+	/// The amount of stagger to apply
+	var/stagger_duration = 3 SECONDS
+	/// The amount of hallucination to apply
+	var/hallucination_duration = 25 SECONDS
+	/// Helmet the spell's bound to.
+	var/obj/item/clothing/head/helmet/perceptomatrix/linked_helmet
+	/// Spark system
+	var/datum/effect_system/spark_spread/quantum/spark_sys
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/New(Target)
+	. = ..()
+
+	linked_helmet = target
+	spark_sys = new /datum/effect_system/spark_spread/quantum
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/Destroy()
+	linked_helmet = null
+	QDEL_NULL(spark_sys)
+	return ..()
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/is_valid_target(atom/cast_on)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(ishuman(cast_on))
+		return TRUE
+	if(istype(cast_on, /obj/item/food/pancakes))
+		return TRUE
+
+	return FALSE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/can_cast_spell(feedback = TRUE)
+	. = ..()
+	if(!.)
+		return .
+
+	if(!linked_helmet)
+		stack_trace("casting w/o linked perceptomatrix!")
+
+	return .
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/proc/blows_up_pancakes_with_mind(obj/item/food/pancakes/pancakes)
+	cast_fx(pancakes)
```

Comment:
You can just do the cast fx in `cast_on` since you play it regardless of the target being human or pancake
```suggestion
```